src/Compilers/CSharp/Portable/Parser/DirectiveParser.cs

```diff
@@ -308,7 +309,24 @@ private DirectiveTriviaSyntax ParseErrorOrWarningDirective(SyntaxToken hash, Syn
                 //could be negative if part of the error text comes from the trailing trivia of the keyword token
                 int triviaOffset = eod.GetLeadingTriviaWidth() - triviaWidth;
 
-                eod = this.AddError(eod, triviaOffset, triviaWidth, isError ? ErrorCode.ERR_ErrorDirective : ErrorCode.WRN_WarningDirective, triviaBuilder.ToString());
+                string errorText = triviaBuilder.ToString();
+                eod = this.AddError(eod, triviaOffset, triviaWidth, isError ? ErrorCode.ERR_ErrorDirective : ErrorCode.WRN_WarningDirective, errorText);
+
+                if (isError)
+                {
+                    if (errorText.Equals("version", StringComparison.Ordinal))
+                    {
+                        Assembly assembly = typeof(DirectiveParser).GetTypeInfo().Assembly;
```

Comment:
I think this should be typeof(CSharpCompiler) to match the type used to get the assembly in the compiler. It might happen to be the same assembly today, but better not to make the code depend on that fact.